bin/node/src/aleph_node_rpc.rs

```diff
@@ -156,6 +168,11 @@ pub trait AlephNodeApi<BE> {
     ///
     #[method(name = "ready")]
     fn ready(&self) -> RpcResult<bool>;
+
+    #[method(name = "validatorNetworkInfo")]
+    async fn validator_network_info(
```

Comment:
There is a notion of `Safe` and `Unsafe` RPC calls in substrate, and how to tell which is which was always a big mystery to me. Anyway, the `system_unstable_networkState` is unsafe, probably because it might output lots of data. I'm wondering whether we could/should make this unsafe too. Being unsafe has this disadvantage that our RPC nodes have only safe methods exposed, so things might be annoying :|

Reply:
I think marking it "Unsafe" is a good idea, we haven't analyzed the data returned for vulnerabilities (it's mostly public, but maybe something can be inferred from which exact data is cached?) and no one who cannot run their own node would even be interested in the output anyway. This will make it slightly less convenient to use, but it shouldn't be that much of a problem?

Reply:
Now marked as unsafe. Substrate says that `system_unstable_networkState` is unsafe mainly because its output api is not final yet, but they don't work on stabilizing it actively for quite some time.
I tried not to add new flags to cli, so right now that logic is run only for non-validators with rpc enabled.

Reply:
In our infra:
- RPC nodes (archivists) have only Safe RPC methods enabled,
- validators have also Unsafe methods enabled (this is safe, because they are not reachable from outside anyway)

I think ideally we would like this to be possible to call on at least some of nodes in our infra :P And from what you are saying 
> right now that logic is run only for non-validators with rpc enabled.

it would not?

Reply:
You're right. Changed to be run only on nodes with `rpc` and `telemetry` enabled, and while the latter is not strongly connected to this, it is at least partially imho and can be switched off from `cli`. 
Ideally, I think we should have a separate config file for all such values and consts that have too specific role to have a separate `cli` flag, but can be sometimes useful to be modified. I'll consult what the rest of the team thinks, but even if we agree to that solution, it would be a separate task.

Reply:
I think `telemetry` is quite unrelated, let's not mix it in, because that adds a strange constraint. 

The other thing I'm not quite sure is:

```
   let validator_address_cache = match (&config.role, config.telemetry_endpoints.is_some()) {
        (sc_network::config::Role::Full, true) => Some(ValidatorAddressCache::new()),
```
above what does it really mean `sc_network::config::Role::Full`, i.e., for instance in our infra:
- are our archivists full?
- are our validators full?

Reply:
So the only option I see is that I'm adding a cli flag. 
Also removed criterion based on `Role`, but fyi there are two roles:
* Role::Authority for nodes that can participate in consensus (our validators)
* Role::Full for nodes that don't (our archivists)

It's not that easy however, because there is also another struct named `ObservedRole`, which refers to other nodes from our point of view, and it also includes light role (like light clients). To my understanding, we don't care about us being light clients there because `aleph-node` codebase is not supposed to support being light clients (we'd need a separate repo for that).

Reply:
I think the current version is good, but there is one downside of having an additional flag -- a change must be introduced to our infra as well, which imho makes it troublesome. So there are two options:
1. Get rid of the flag, and enable this for all nodes. 
2. Keep the current version, but then need to coordinate with infra, on what changes must be made along the release of this. 

If I had to decide, I would just go with 1. because it's simplest, and the overhead is negligible anyway -- @timorleph @kostekIV ?

Reply:
Option 1 sounds good enough for me although imho 2 would be ideal

Reply:
Hmm, how about option 1.5: leave the flag, but enable by default?

Reply:
Yes, I like it :)

Reply:
Fee free to resolve the comment yourself @ggawryal 

I'm ok with merging